Docs/coverage (#69)
xedi/sdk-js: docs/content/tooling/coverage.mdx

---
title: Coverage
description: How to generate a code coverage report
status: Stable

> source: https://github.com/xedi/sdk-js/tree/master/testing

Reviewer: Is this necessary? I'm torn. I see the value but at the same time it feels out of place... /shrug

---

> import {Text, Flash, BorderBox} from '@primer/components'

Reviewer: ```suggestion
import {Flash} from '@primer/components'
```

import {InfoIcon} from '@primer/styled-octicons'
import InlineCode from '@primer/gatsby-theme-doctocat/src/components/inline-code'
import {Link} from 'gatsby';

When writing tests it can often prove difficult to ensure we have sufficient coverage across all our code. 
In this section we will look at using a coverage report to allow us to determine if further testing is required.

## Generating a report

> It is first important to ensure the tests are passing and to a high standard. read the <Link to="/tooling/testing">testing</Link> section more infomation

Reviewer: ```suggestion
It is first important to ensure the tests are passing and to a high standard. Read the <Link to="/tooling/testing">testing</Link> section more information.
```

Once that is done a coverage report can be generated by simply running: <InlineCode>npm run coverage</InlineCode>.

> <Flash scheme="yellow">

Reviewer: I would be tempted to stick with the default theme.

Blue = info
Yellow = warning
Red = danger

These are the generally accepted semantics of those colours

Author: I saw this as a warning that the code coverage folder would need to be ignored so the choice of yellow was intentional. Are you thinking this is more an info flash?

    <InfoIcon mr={2}/>
    Running a coverage report will create a coverage directory where the generated files are stored. Please ensure this folder has been .gitignored.
</Flash>

## Viewing a report

To see the results of the report you will need to locate the index.html file that will have been generated and placed in the coverage folder and open this in your browser.

## Understanding a report.

Once open you will see a breakdown of the project and an indicator as a percentage of how good the test coverage is. We strive to maintain a code coverage of at least 80%.

> Where codecoverage is not up to par we can click individual sections and drill down further so we are able to easily see which aspects of the code require further testing.

Reviewer: ```suggestion
Where code coverage is not up to that standard we can click individual sections and drill down further so we are able to easily see which aspects of the code require further testing.
```
